### ISSUE TITLE
`isinstance(instance, SubClass)` loses type instead of narrowing

### ISSUE THREAD

**Issue by jeremyBanks** [Microsoft/pyright#24]
Given an `instance` of a `SuperClass`, I would expect `isinstance(instance, SubClass)` to narrow the type of `instance` to `SubClass` in the following branch. Instead, the type appears to have been lost and replaced with `NoneType`.

```python
class SuperClass:
    def __init__(self) -> None:
        self.property: bool = True

class SubClass(SuperClass):
    pass

def function(instance: SuperClass) -> None:
    if isinstance(instance, SubClass):
        print(instance.property)
```

```none
$ mypy --strict example.py && echo "no errors"
no errors
```
```none
$ pyright example.py
Finding source files
Found 1 source files
Analyzed 1 file in 1.229sec
/Users/jbanks/Desktop/example.py
  'property' is not a known member of 'None' (10, 24)
1 error, 0 warnings
```

For a slightly smaller example, we get the same error narrowing from the base `object` type:

```python
class MyClass:
    def __init__(self) -> None:
        self.property: bool = True


def function(instance: object) -> None:
    if isinstance(instance, MyClass):
        print(instance.property)
```

**Comment by erictraut** [comment]
This is fixed in 1.0.4, which I just published. Thanks for the bug report.
